tinyxml2.cpp

```diff
@@ -2304,6 +2282,17 @@ void XMLDocument::Parse()
     ParseDeep(p, 0, &_parseCurLineNum );
 }
 
+template<class NodeType, int PoolElementSize>

```

Comment:
PoolElementSize if also used to pass a pool of the exact derived type and have a chance to avoid a virtual call to Alloc(). Since it's deduced from the parameter I don't see any cognitive overhead.